diff --git a/camkes/parser/stage9.py b/camkes/parser/stage9.py
index eb1607a..d97d672 100644
--- a/camkes/parser/stage9.py
+++ b/camkes/parser/stage9.py
@@ -19,7 +19,6 @@ Combine n-1 connections according to the following rule:
 
 from __future__ import absolute_import, division, print_function, \
     unicode_literals
-from future.utils import iteritems
 from itertools import chain
 from camkes.internal.seven import cmp, filter, map, zip
 
@@ -46,7 +45,7 @@ def add_connections(connections):
 # conn_to_add is a single connection to replace them with
 def consolidate(candidates):
     multi = []
-    for (_, i) in iteritems(candidates):
+    for (_, i) in candidates.items():
         # Combine connection sets if any two sets share a connection
         for j in multi:
             if i & j:
