diff --git a/python-capdl-tool/capdl/ELF.py b/python-capdl-tool/capdl/ELF.py
index 7657945..9faa28c 100644
--- a/python-capdl-tool/capdl/ELF.py
+++ b/python-capdl-tool/capdl/ELF.py
@@ -11,7 +11,6 @@ without elftools installed by not importing this particular file.
 """
 from __future__ import absolute_import, division, print_function, \
     unicode_literals
-from future.utils import iteritems
 
 from elftools.elf.elffile import ELFFile
 from elftools.elf.constants import P_FLAGS
@@ -149,7 +148,7 @@ class ELF(object):
         existing_pages = []
         if addr_space:
             # Update symbols with their vaddrs in the AddressSpaceAllocator if we were given one
-            for (symbol, (sizes, caps)) in iteritems(addr_space.get_symbols_and_clear()):
+            for (symbol, (sizes, caps)) in addr_space.get_symbols_and_clear().items():
                 assert self.get_symbol_size(symbol) >= sum(sizes), \
                     "Symbol (%s:%d) must have same or greater size than supplied cap range (%d)" % (
                         symbol, self.get_symbol_size(symbol), sum(sizes))
