diff --git a/src/plat/qemu-arm-virt/config.cmake b/src/plat/qemu-arm-virt/config.cmake
index af98c16e938..1b81b6e26d9 100644
--- a/src/plat/qemu-arm-virt/config.cmake
+++ b/src/plat/qemu-arm-virt/config.cmake
@@ -107,7 +107,7 @@ if(KernelPlatformQEMUArmVirt)
                 string(
                     REGEX
                         MATCH
-                        "[0-9](\\.[0-9])+"
+                        "[0-9]+(\\.[0-9]+)+"
                         QEMU_VERSION
                         "${QEMU_STDOUT_MESSAGE}"
                 )
diff --git a/src/plat/qemu-riscv-virt/config.cmake b/src/plat/qemu-riscv-virt/config.cmake
index de62061d635..af6016b89e7 100644
--- a/src/plat/qemu-riscv-virt/config.cmake
+++ b/src/plat/qemu-riscv-virt/config.cmake
@@ -44,7 +44,7 @@ if(KernelPlatformQEMURiscVVirt)
                 string(
                     REGEX
                         MATCH
-                        "[0-9](\\.[0-9])+"
+                        "[0-9]+(\\.[0-9]+)+"
                         QEMU_VERSION
                         "${QEMU_STDOUT_MESSAGE}"
                 )

