diff --git a/gcc.cmake b/gcc.cmake
index 7636cab1..3f018f71 100644
--- a/seL4-src/gcc.cmake
+++ b/seL4-src/gcc.cmake
@@ -104,6 +104,7 @@ if("${CROSS_COMPILER_PREFIX}" STREQUAL "")
                 CROSS_COMPILER_PREFIX
                 "riscv64-unknown-linux-gnu-"
                 "riscv64-unknown-elf-"
+                "riscv64-none-elf-"
                 "riscv64-elf-"
             )
         endif()
